simtools/data_model/model_data_writer.py

```diff
@@ -113,33 +95,40 @@ def validate_and_transform(self, product_data=None, validate_schema_file=None):
         )
         return _validator.validate_and_transform()
 
-    def write_data(self, product_data):
+    def write(self, product_data=None, metadata=None):
         """
-        Write model data.
+        Write model data and metadata
 
         Parameters
         ----------
         product_data: astropy Table
-            Model data to be written.
+            Model data to be written
+        metadata: dict
+            Metadata to be written.
 
         Raises
         ------
         FileNotFoundError
-            if data writing was not sucessfull.
+            if data writing was not successful.
 
         """
 
+        if product_data is None:
+            return
```

Comment:
I would prefer to include a warning here to raise the awareness that nothing was written (although one used a `write` function). To be honest, I did not understand the reasoning behind it. Why not making `product_data` a required argument?

Reply:
One wouldn't need an if statement if the argument is required, it would be actually less code than the current version. Personally, I wouldn't like if I run a function that is supposed to write something and I don't get something written in the end without being told about it. But if you think this is style, then we can go ahead.